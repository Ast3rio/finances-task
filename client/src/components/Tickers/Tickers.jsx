import React, {useState} from 'react';
import styles from './Tickers.module.scss';
import Ticker from "./Ticker";
import {Button} from "../Button/Button";

export const Tickers = ({tickers}) => {

    const [onSort, setOnSort] = useState(false);

    return (
        <div className={styles.tickers}>
            <Button
                label='Sort by Price'
                active={onSort}
                onClick={() => setOnSort(current => !current)}
            />
            <ul className={styles.list}>
                {tickers.sort((a, b) => onSort && a.price < b.price ? 1 : -1)
                    .map(({ticker, ...tickers}) =>
                        <Ticker key={ticker} ticker={ticker} {...tickers} />
                    )}
            </ul>
        </div>
    );
};
