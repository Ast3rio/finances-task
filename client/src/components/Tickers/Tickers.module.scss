
.tickers{
  width: 800px;
  min-height: 400px;
  margin: 50px auto;
}

.list{
  width: 100%;
  height: 400px;
  margin-top: 24px;
  overflow: auto;
  transition: all .5s;

}

