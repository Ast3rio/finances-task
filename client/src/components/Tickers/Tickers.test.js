import React from "react";
import {screen, render} from '@testing-library/react';
import { Tickers } from './Tickers';

describe('Tickers', () => {
    it('render Tickers component', () => {
        render(<Tickers />)
        screen.getByRole('button')
        screen.debug()
        screen.getByRole('list')
        screen.debug()
    })
})
