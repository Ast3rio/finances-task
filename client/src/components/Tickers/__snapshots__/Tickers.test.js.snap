// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Tickers/> Tickers have renderer correctly 1`] = `ShallowWrapper {}`;
